onyo/commands/get.py

```diff
@@ -0,0 +1,145 @@
+import logging
+import re
+import sys
+from collections import Counter
+from typing import Generator, Union
+from pathlib import Path
+
+from rich import box
+from rich.console import Console
+from rich.table import Table
+
+from onyo import Filter, Repo, OnyoInvalidRepoError, OnyoInvalidFilterError
+
+logging.basicConfig()
+log = logging.getLogger('onyo')
+
+
+def natural_sort(
+        assets: list[tuple[Path, dict[str, str]]],
+        keys: Union[list, None] = None, reverse: bool = False) -> list:
+    """
+    Sort the output of `Repo.get()` by a given list of `keys`.
+    """
+    if not keys:
+        assets = sorted(
+            assets,
+            key=lambda x: [
+                int(s) if s.isdigit() else s.lower()
+                for s in re.split('([0-9]+)', str(x[0]))],
+            reverse=reverse)
+    else:
+        for key in reversed(keys):
+            assets = sorted(
+                assets,
+                key=lambda x: [
+                    int(s) if s.isdigit() else s.lower() for s in
+                    re.split('([0-9]+)', str(x[1][key]))],
+                reverse=reverse)
+
+    return assets
+
+
+def fill_unset(
+        assets: Generator[tuple[Path, dict[str, str]], None, None],
+        keys: list, unset: str = '<unset>') -> Generator:
+    """
+    If a key is not present for an asset, define it as `unset` (default is
```

Comment:
An asset having a field but no value set (`key: `) is invalid yaml, so you wouldn't be able to save (or load) this with `ruamel.yaml`. It is possible to use a null value (e.g., `key: null`). I consider this as unset, although I return a `NoneType` as a key value in the response. The use of `null` is very specific and becomes `None` in Python.

Reply:
`key: ` is by definition invalid, but `ruamel.yaml` does allow this to happen, which is a bug. We have an issue about the need for better YAML syntax checks: https://github.com/psyinfra/onyo/issues/117
I added this case to the issue, so that we do not forget.

`null` values are not adressed in `onyo set`, this whole class of unset values, or adding keys without values brings some problems with it, which is why we left it out. You just jumped into it together with `onyo get`, but I am as of yet unsure how to do this correctly in a unified way all along onyo ;)

Reply:
I agree that we should do our best to avoid that `ruamel.yaml` bug. As for `<unset>`, it's primary intended purpose is to get assets where a key is unset which is what it does. If you try to read an asset using `ruamel.yaml` where there is an empty `key: `, `ruamel.yaml` will raise an error (I tested this locally at least). Thus any such checking should be done at the reading and writing stage, and `get`, beyond its own use of `Repo._read_asset()` should not be confronted with invalid keys.

Reply:
I can not reproduce this locally, for me it works without error to read and to write.

That said, it is true that it should be handled by `Repo._read_asset()`.